tools/code_format/check_format.py

```diff
@@ -16,232 +16,120 @@
-    "Protobuf::NULL_VALUE":             "ProtobufWkt::NULL_VALUE",
-    "Protobuf::StringValue":            "ProtobufWkt::StringValue",
-    "Protobuf::Struct":                 "ProtobufWkt::Struct",
-    "Protobuf::Value":                  "ProtobufWkt::Value",
-
-    # Other common mis-namespacing of protobuf types.
-    "ProtobufWkt::Map":                 "Protobuf::Map",
-    "ProtobufWkt::MapPair":             "Protobuf::MapPair",
-    "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
-}
-# yapf: enable
-
-LIBCXX_REPLACEMENTS = {
-    "absl::make_unique<": "std::make_unique<",
-}
-
-CODE_CONVENTION_REPLACEMENTS = {
-    # We can't just remove Times(1) everywhere, since .Times(1).WillRepeatedly
-    # is a legitimate pattern. See
-    # https://github.com/google/googletest/blob/master/googlemock/docs/for_dummies.md#cardinalities-how-many-times-will-it-be-called
-    ".Times(1);": ";",
-    # These may miss some cases, due to line breaks, but should reduce the
-    # Times(1) noise.
-    ".Times(1).WillOnce": ".WillOnce",
-    ".Times(1).WillRepeatedly": ".WillOnce",
-    "Stats::ScopePtr": "Stats::ScopeSharedPtr",
-}
-
-UNSORTED_FLAGS = {
-    "envoy.reloadable_features.activate_timers_next_event_loop",
-    "envoy.reloadable_features.grpc_json_transcoder_adhere_to_buffer_limits",
-}
+# The way this script is currently used (ie no bazel) it relies on system deps.
+# As `pyyaml` is present in `envoy-build-ubuntu` it should be safe to use here.
+import yaml
 
 logger = logging.getLogger(__name__)
 
-LINE_NUMBER_RE = re.compile(r"^(\d+)[a|c|d]?\d*(?:,\d+[a|c|d]?\d*)?$")
-VIRTUAL_INCLUDE_HEADERS_RE = re.compile(r"#include.*/_virtual_includes/")
-OWNER_RE = re.compile('@\S+')
-PROJECT_OWNERS_RE = re.compile(r'.*github.com.(.*)\)\)')
+
+class FormatConfig:
+
+    def __init__(self, path: str) -> None:
+        self.path = path
+
+    def __getitem__(self, k):
+        return self.config.__getitem__(k)
+
+    @cached_property
+    def buildifier_path(self) -> str:
+        return paths.get_buildifier()
+
+    @cached_property
+    def buildozer_path(self) -> str:
+        return paths.get_buildozer()
+
+    @cached_property
+    def clang_format_path(self) -> str:
+        return os.getenv("CLANG_FORMAT", "clang-format-14")
+
+    @cached_property
+    def config(self) -> Dict:
+        return yaml.safe_load(pathlib.Path(self.path).read_text())
+
+    @cached_property
+    def paths(self) -> Dict[str, Tuple[str, ...] | Dict[str, Tuple[str, ...]]]:
+        paths = self._normalize("paths", cb=lambda paths: tuple(f"./{p}" for p in paths))
+        paths["build_fixer_py"] = self._build_fixer_path
+        paths["header_order_py"] = self._header_order_path
+        return paths
+
+    @cached_property
+    def re(self) -> Dict[str, Pattern[str]]:
+        return {k: re.compile(v) for k, v in self["re"].items()}
+
+    @cached_property
+    def re_multiline(self) -> Dict[str, Pattern[str]]:
```

Comment:
Can you add some comments to help readability for these properties?

Reply:
docstrings added and the erroneously named `subdir_set` has been shifted to config and renamed